CMakeLists.txt

```diff
@@ -18,10 +18,87 @@ elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
     if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.3)
         message(FATAL_ERROR "Clang version must be at least 3.3!")
     endif()
+elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
+    # allow MSVC VS2015.1 and VS2013 toolchain ( VS2013 is not tested )
+    if ("${CMAKE_CXX_COMPILER_VERSION}" VERSION_EQUAL "19.0")
+        if (NOT CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL "19.0.23506.0")
```

Comment:
I'm not familiar with the windows world, but does it make sense to require such a specific version number?

Reply:
This is the exact version of MS Visual C++ Compiler that comes with MS Visual Studio 2015 Update 1, which is to my knowledge the only version of MSVC that is able to compile the codebase.

Reply:
FYI: Support for the latest versions of MSVC are in code review.

Reply:
That sounds great. Would it also support 2015, or 2017 only ? Or do you mean only the freshest 2019 ?

Reply:
I have not downloaded 2019 yet, but my guess is that it will work. I am still testing.